### PR TITLE
Refactor for reusability

### PR DESCRIPTION
**Pull request by sw3dish**
This commit tries to remove some unused/outdated code and refactor the code to make it more reusable.

To test:
- Checkout master branch and run the script
  - Run pipenv shell
- Checkout this branch and run the script
  - Run pipenv shell
- Make sure that diff does not subtract anything from the file generated from the master branch
  - There should be at least a header line that exists in the file created by this branch that differs

### CODE REVIEW

**Comment by wbrown22** [comment]
L.G.T.M

**Comment by righteousrenegade** [comment]
Could we add the pipenv instructions in the README? I just ended up defaulting to my standard virtualenv process. Might be useful to add the pipenv instructions into the readme to indicate that that's our preferred way of handling dependencies for this repo.

also +1 for pipenv.

File: dive_authors.py [hunk 1]
```diff
@@ -191,37 +145,85 @@ def get_authordateinfo_list(self):
...
+    repos = []
+    page = 1
+    url = 'https://api.github.com/orgs/industrydive/repos?per_page=100&page=%s&access_token=%s'
+    while True:
+        response = requests.get(url % (page, access_token))
+
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            print 'Error: %s' % e
```
Comment (righteousrenegade) [inline dive_authors.py]: i think using `print` statements is generally frowned on. can we use some kind of logger and pipe it to stdout or something?

File: dive_authors.py [hunk 2]
```diff
@@ -191,37 +145,85 @@ def get_authordateinfo_list(self):
...
-        repo_path = os.path.join(projects_path, project_name)
+            repos = get_all_repos(access_token)
 
-        runner = DiveRunner()
-        runner.repo = repo_path
-        runner.project_name = project_name
+            for repo in repos:
+                # if the repo has been pushed to after the first day of the target year
+                pushed_at_time = datetime.datetime.strptime(repo['pushed_at'], '%Y-%m-%dT%H:%M:%SZ')
+                if pushed_at_time > datetime.datetime(year, 1, 1):
```
Comment (righteousrenegade) [inline dive_authors.py:213]: should this be `>=` instead of `>`? i think the way it's written it would only get things pushed on or after Jan. 2, right?

**Comment by righteousrenegade** [comment]
OK, looks good to me, and works as far as i can tell. I did have like, 2 things i commented that i think need changing, but other than that, i think it's fine.

 I did get 1574 lines in the result from the refactor, comopared to 1558 from the branch on master. I think that's likely because previously we were explicitly listing the repos though whereas this one is automatically figuring out what all our repos are, so i dont think that number discrepancy is a concern.

File: dive_authors.py (hunk 2 above)
Comment (sw3dish) [reply dive_authors.py:213]: `datetime.datetime(2019, 1, 1)` creates an object that is 2019-01-01 00:00:00, so it should get everything published before midnight. I guess I should add the `=` anyway though

**Comment by sw3dish** [comment]
@jlegs yeah I noticed that as well -- I think it pulls in some lines of 0 additions, 0 deletions. But I don't know why it's doing that because I didn't refactor any of the actual code that generates the lines (apart from removing what was essentially an `if True:` condition)

File: dive_authors.py (hunk 1 above)
Comment (sw3dish) [reply dive_authors.py]: Yah, using `logging` now

**Comment by righteousrenegade** [comment]
OK merging this.

**Pull request merged**